src/main/java/com/byclosure/jenkins/plugins/gcloud/GCloudBuildWrapper.java

```diff
@@ -51,11 +102,25 @@ public boolean tearDown(AbstractBuild build, final BuildListener listener) throw
 				serviceAccount.cleanUp();
 				return true;
 			}
-
 		};
 	}
 
-	public String getCredentialsId() {
+    public GCloudInstallation getSDK() {
+        GCloudInstallation[] installations = GCloudInstallation.getInstallations();
+        if (installations.length == 1) return installations[0];
+        for (GCloudInstallation sdk : installations) {
+            if (installation.equals(sdk.getName())) return sdk;
+        }
+        throw new IllegalArgumentException("Invalid GCloud SDK installation "+installation);
```

Comment:
IMO the method should return null in such case. There're many potential use-cases of the method, such exception-based approach is not handy.

P.S: throwing of undocumented Runtime exceptions is a really bad practice